cloudify_agent/shell/commands/configure.py

```diff
@@ -73,53 +77,100 @@ def _disable_requiretty():
     It is needed in case the environment was created using different paths
     than the one that is used at runtime.
 
     """
+    # this code is mostly taken from virtualenv; we can't use virtualenv's
+    # make_environment_relocatable because it checks if the old shebang
+    # points to the virtualenv dir, and packaged agents use a non-existent
+    # /tmp path. So, we copied _fixup_scripts mostly intact, but disabled
+    # the shebang checking.
+    _make_environment_relocatable(VIRTUALENV)
+
+
+def _make_environment_relocatable(home_dir):
+    home_dir, lib_dir, inc_dir, bin_dir = path_locations(home_dir)
+
+    _fixup_scripts(bin_dir)
+    fixup_pth_and_egg_link(home_dir)
+
 
+def _fixup_scripts(bin_dir):
+    """Make scripts in bin_dir relative by rewriting their shebangs
+
+    Examine each file in bin_dir - if it looks like a python script, and has a
+    shebang - replace it with a new, "relative" shebang. (like
+    `virtualenv --relocatable` would)
+
+    The relative shebang is platform-specific, but on linux it will consist
+    of a /usr/bin/env shebang, and a python snippet that runs the `activate`
+    script.
+    """
     from cloudify_agent.shell.main import get_logger
     logger = get_logger()
 
-    bin_dir = '{0}/bin'.format(VIRTUALENV)
+    new_shebang = _get_relative_shebang()
+    for filename in _find_scripts_to_fix(bin_dir):
+        logger.info('Making script %s relative' % filename)
+        _rewrite_shebang(filename, new_shebang)
+
 
-    logger.debug('Searching for executable files in {0}'.format(bin_dir))
-    for executable in os.listdir(bin_dir):
-        path = os.path.join(bin_dir, executable)
-        logger.debug('Checking {0}...'.format(path))
-        if not os.path.isfile(path):
-            logger.debug('{0} is not a file. Skipping...'.format(path))
+def _find_scripts_to_fix(bin_dir):
+    """Search bin_dir for files that look like python scripts with a shebang
+    """
+    for filename in os.listdir(bin_dir):
+
+        if filename in OK_ABS_SCRIPTS:
             continue
-        if os.path.islink(path):
-            logger.debug('{0} is a link. Skipping...'.format(path))
+
+        filename = os.path.join(bin_dir, filename)
+        if not os.path.isfile(filename):
+            # ignore subdirs, e.g. .svn ones.
             continue
-        basename = os.path.basename(path)
-        if basename in ['python', 'python2.7', 'python2.6']:
-            logger.debug('{0} is the python executable. Skipping...'
-                         .format(path))
+
+        with open(filename, 'rb') as f:
+            try:
+                lines = f.read().decode('utf-8').splitlines()
+            except UnicodeDecodeError:
+                # This is probably a binary program instead
+                # of a script, so just ignore it.
+                continue
+
+        if not lines:
+            continue
+
+        shebang = lines[0]
+        if not (shebang.startswith('#!') and shebang.endswith('python')):
```

Comment:
yes, eh, there's this tradeoff here, between being more general, and needing to assume more things :D

well, the number of possibilities there is very limited so maybe let's not overthink this: i think just checking `"bin/python" in path` would cover all cases while not causing any false positives in realistic scripts?